### PR TITLE
Task details modal implemented

### LINKED ISSUE #226: Change Task Details to Modal Dialog instead of own route

### PR DESCRIPTION
**Pull request by YohannesTesfay**
Closes #226

### CODE REVIEW

**Comment by YohannesTesfay** [comment]
I deleted and moved some files/ folders. The package.json is an update on the browser-list that usually pops up when we use `npm run dev`.

File: app/frontend/components/taskDetailsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        data: {
+          projectId: taskData.projectId,
+          title: taskData.title,
+        },
+      })
+      if (result.error) {
+        throw new Error(`GraphQL Error ${result.error}`)
+      }
+      reset()
+    } catch {}
```
Comment (PascalHelbig) [inline app/frontend/components/taskDetailsModal.tsx:36]: add a `console.error` in catch block.

File: app/frontend/components/taskList/taskList.tsx [hunk 2]
```diff
@@ -23,27 +23,20 @@ export interface TaskListProps {
...
   const {
     register,
     handleSubmit,
     reset,
     formState: { isSubmitting, errors },
-  } = useForm<TaskForm>()
+  } = useForm<TaskInput>()
   const [, taskCreate] = useTaskCreateMutation()
   const [taskToBeDeleted, setTaskToBeDeleted] = useState<TaskFragment | undefined>()
+  const [currentTask, setCurrentTask] = useState<TaskFragment | undefined>()
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskList.tsx:37]: rename state to `taskToBeUpdated`

File: app/frontend/components/taskList/taskList.tsx [hunk 3]
```diff
@@ -59,65 +52,76 @@ export const TaskList = (props: TaskListProps): JSX.Element => {
...
-              </Button>
-              <span className="ml-2">{task.title}</span>
-            </TableCell>
-            <TableCell className="text-center">
-              <input type="checkbox" />
-            </TableCell>
+    <>
+      {currentTask ? (
+        // eslint-disable-next-line unicorn/no-useless-undefined
+        <TaskDetailsModal open task={currentTask} onClose={() => setCurrentTask(undefined)} />
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskList.tsx:58]: remove `open` prop from TaskDetailsModal (here + definition)

Comment (PascalHelbig) [inline app/frontend/components/taskList/taskList.tsx:58]: Move this modal to the other delete modal.

**Review by PascalHelbig — APPROVED** [review]

**Pull request merged**